sql/python/sqlflow/template/db.py

```diff
@@ -0,0 +1,177 @@
+
+    def execute(self, query):
+        """ Execute the query
+        Args:
+            query(str): the query to execute
+
+        Returns:
+            cursor(cursor):
+        """
+        cursor = self._db.cursor()
+        cursor.execute(query)
+        return cursor
+
+    def execute_many(self, query, values):
+        """ Execute the query with values
+        Args:
+            query(str): the query to execute
+            values(list(tuple)): the values
+
+        Returns:
+            cursor(cursor)
+        """
+        cursor = self._db.cursor()
+        cursor.executemany(query, values)
+        return cursor
+
+    def create_table(self, table_name, table_schema):
+        """ Create table
+        Args:
+            table_name(str): name of table
+            table_schema(list(tuple)): schema of table
+
+        Returns:
+            success if no error thrown out
+        """
+        statement = self._gen_create_table_statement(table_name, table_schema)
+        self.execute(statement)
+
+    def insert_values(self, table_name, table_schema, values):
+        """ Insert values to table
+        Args:
+            table_name(str): name of table
+            table_schema(list(str)): schema of table
+            values(list(tuple)): the value to insert
+
+        Returns:
+            success if no error thrown out
+        """
+        statement = self._gen_insert_table_statement(table_name, table_schema)
+        self.execute_many(statement, values)
+
+    def query_select(self, statement):
+        """ Select fields from table simply
+        Args:
+            statement(str): select statement
+
+        Return:
+            Returns:
+            field_names(list(str)): the names of fields
+            field_data(list): return data of column-style
+        """
+        cursor = self.execute(statement)
+        return self._retrieve_fields_and_value_from_cursor(cursor)
+
+    def _connect(self):
+        raise NotImplementedError("DataBase.connect")
+
+    def _gen_create_table_statement(self, table_name, table_schema):
+        # TODO replace the type of fields according to different kind of database
+        return '''CREATE TABLE `{}` ({})'''.format(
+            table_name, ", ".join(["%s %s" % (n, t) for n, t in table_schema]))
+
+    def _gen_insert_table_statement(self, table_name, table_schema):
+        raise NotImplementedError("DataBase._gen_insert_table_statement")
+
+    def _retrieve_fields_and_value_from_cursor(self, cursor):
+        raise NotImplementedError("DataBase._retrieve_fields_and_value_from_cursor")
+
+
+class MySQLDataBase(DataBase):
+
+    def _connect(self):
+        from mysql.connector import connect
+        return connect(user=self._user,
+                       passwd=self._password,
+                       database=self._database,
+                       host=self._host,
+                       port=self._port)
+
+    def _retrieve_fields_and_value_from_cursor(self, cursor):
```

Comment:
`_retrieve_fields_and_value_from_cursor` I found this function is exactly the same across `MySQLDataBase`, `SQLite3DataBase` and `HiveDataBase`...

Reply:
No, `HiveDataBase ` is different

Reply:
My bad.